internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicMetadataFetcher.java

```diff
@@ -150,22 +155,25 @@ class TopicMetadataFetcher implements Closeable {
   }
 
   // acquire the consumer from the pool
-  private PubSubConsumerAdapter acquireConsumer() {
+  PubSubConsumerAdapter acquireConsumer() {
     try {
-      return pubSubConsumerPool.take();
+      consumerWaitListSize.incrementAndGet();
+      long startTime = System.nanoTime();
+      PubSubConsumerAdapter consumerAdapter = pubSubConsumerPool.take();
+      stats.recordLatency(CONSUMER_ACQUISITION_WAIT_TIME, startTime);
+      return consumerAdapter;
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new VeniceException("Interrupted while acquiring pubSubConsumerAdapter", e);
+    } finally {
+      consumerWaitListSize.decrementAndGet();
     }
   }
 
   // release the consumer back to the pool
-  private void releaseConsumer(PubSubConsumerAdapter pubSubConsumerAdapter) {
-    try {
-      pubSubConsumerPool.put(pubSubConsumerAdapter);
-    } catch (InterruptedException e) {
-      Thread.currentThread().interrupt();
-      throw new VeniceException("Interrupted while releasing pubSubConsumerAdapter", e);
+  void releaseConsumer(PubSubConsumerAdapter pubSubConsumerAdapter) {
+    if (!pubSubConsumerPool.offer(pubSubConsumerAdapter)) {
+      LOGGER.error("Failed to release pubSubConsumerAdapter back to the pool.");
```

Comment:
Is it possible that the `offer` can fail? As the total number of consumers is fixed, so in theory, it shouldn't fail.

Reply:
yeah it shouldn't fail unless there are some logical bugs in the code.